controllers/platformnetwork_controller.go

```diff
@@ -661,37 +698,220 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
+			}
 		}
+	} else {
+		network, err := r.FindExistingNetwork(client, instance)
+		if err != nil {
+			return false, err
+		} else if network == nil {
+			return true, nil
+		}
+		if _, ok := plNetworkUpdateRequired(instance, network); ok {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (r *PlatformNetworkReconciler) PlatformNetworkUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	pool_update_required, err := r.AddressPoolUpdateRequired(client, instance)
+	if err != nil {
+		err = common.NewSystemDependency("There was an error retrieving the address pool.")
+		return false, err
+	}
+
+	nwk_update_required, err := r.NetworkUpdateRequired(client, instance)
+	if err != nil {
+		err = common.NewSystemDependency("There was an error retrieving the network.")
+		return false, err
+	}
+
+	return pool_update_required || nwk_update_required, err
+}
 
+func (r *PlatformNetworkReconciler) UpdateInsyncStatus(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork, err error) (bool, error) {
+	inSync := err == nil
+	if inSync != instance.Status.InSync || inSync {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
 		instance.Status.InSync = inSync
+		if inSync && !instance.Status.Reconciled {
+			// Record the fact that platform network has been reconciled atleast once
+			instance.Status.Reconciled = true
+		}
+		err2 := r.Client.Status().Update(context.TODO(), instance)
+		if err2 != nil {
+			logPlatformNetwork.Error(err2, "failed to update platform network status")
+			return inSync, err2
+		}
+	}
+	return inSync, nil
+}
 
-		if !instance.Status.Reconciled {
-			// Record the fact that we have reached inSync at least once.
-			instance.Status.Reconciled = instance.Status.InSync
+func (r *PlatformNetworkReconciler) TriggerHostNetworkReconciliation(inSync bool, nwk_type string, host *hosts.Host, strategy string) bool {
+	if inSync {
+		logPlatformNetwork.Info(fmt.Sprintf("PlatformNetwork reconciliation successful for %s. Triggering host reconciliation.", nwk_type))
+		host_strategy := cloudManager.HostStrategyInfo{
+			StrategyRequired: strategy,
+			Host:             *host,
 		}
+		_ = r.CloudManager.SendHostReconciliationTrigger(host_strategy)
+		return true
+	}
+	return false
+}
+
+func (r *PlatformNetworkReconciler) ReconcileAddressPoolAndNetwork(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
+	err = r.ReconcileAddressPool(client, instance)
+	if err == nil {
+		if networkResourceRequired(instance) {
+			err = r.ReconcileNetwork(client, instance)
+		}
+	}
+	return err
+}
+
+// ReconcileResource interacts with the system API in order to reconcile the
+// state of a data network with the state stored in the k8s database.
+func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
```

Comment:
I still believe this can be further modularized.

Reply:
> I still believe this can be further modularized.

Done.